modules/apps/product-navigation/product-navigation-user-personal-bar-web/src/main/resources/META-INF/resources/view.jsp

```diff
@@ -57,5 +57,103 @@
 		<span class="sign-in text-default" role="presentation">
 			<aui:icon cssClass="sign-in text-default" data="<%= anchorData %>" image="user" label="sign-in" markupView="lexicon" url="<%= themeDisplay.getURLSignIn() %>" />
 		</span>
+
+		<aui:script>
+			(function () {
+				var signInLink = document.querySelector('.sign-in > a');
+
+				if (!signInLink || signInLink.dataset.redirect !== 'false') {
+					return;
+				}
+
+				var signInURL = '<%= themeDisplay.getURLSignIn() %>';
+
+				var modalSignInURL = Liferay.Util.addParams(
+					'windowState=exclusive',
+					signInURL
+				);
+
+				var setModalContent = function (html) {
+					var fragment = document.createRange().createContextualFragment(html);
+
+					var modalBody = document.querySelector('.liferay-modal-body');
+
+					if (modalBody) {
+						modalBody.innerHTML = '';
+
+						modalBody.appendChild(fragment);
+					}
+				};
+
+				var loading = false;
+				var redirect = false;
+				var html = '';
+				var modalOpen = false;
+
+				var fetchModalSignIn = function () {
+					loading = true;
+
+					Liferay.Util.fetch(modalSignInURL)
+						.then(function (response) {
+							return response.text();
+						})
+						.then(function (response) {
+							loading = false;
+
+							if (!response) {
+								redirect = true;
+
+								return;
+							}
+
+							html = response;
+
+							if (modalOpen) {
+								setModalContent(response);
+							}
+						})
+						.catch(function () {
+							redirect = true;
+						});
+				};
+
+				signInLink.addEventListener('mouseover', function (event) {
+					if (loading || html) {
+						return;
+					}
```

Comment:
Instead of repeating this `if` in both listeners, you could move it inside `fetchModalSignIn()`.